testing/install_external_dependencies.sh

```diff
@@ -35,15 +35,28 @@ fi
 
 # Install Python dependencies for Python specific tests
 if [[ -n "$PYTHON" ]] ; then
-  wget https://repo.continuum.io/miniconda/Miniconda${PYTHON}-4.2.12-Linux-x86_64.sh -O miniconda.sh
-  bash miniconda.sh -b -p $HOME/miniconda
+  case "$(uname)" in
+  Darwin)
+    os=MacOSX
+    ;;
+  *)
+    os=Linux
+    ;;
+  esac
+  if [ ! -e miniconda.sh ]; then
+    wget "https://repo.continuum.io/miniconda/Miniconda${PYTHON}-4.2.12-$os-$(uname -m).sh" -O miniconda.sh.part
+    mv miniconda.sh.part miniconda.sh
+  fi
+  if [ ! -e "$HOME/miniconda" ]; then
+    bash miniconda.sh -b -p "$HOME/miniconda"
+  fi
   echo "export PATH='$HOME/miniconda/bin:$PATH'" >> ~/.environ
   source ~/.environ
   hash -r
   conda config --set always_yes yes --set changeps1 no
   conda update -q conda
   conda info -a
   conda config --add channels conda-forge
-  conda install -q matplotlib pandasql ipython=5.4.1 jupyter_client ipykernel matplotlib bokeh=0.12.10
+  conda install -q matplotlib pandasql ipython=5.4.1 jupyter_client ipykernel matplotlib bokeh=0.12.10 py4j
```

Comment:
not sure we should install py4j here - spark for example carries its own version of it

Reply:
Local tests didn't work without it (even though I have Spark installed). But I don't want to mess up Travis setup, so I will add a separate "if" statement for it.